### DISCUSSION TITLE
Send custom Emails/Notifications

### DISCUSSION THREAD

**Discussion by mffap** [zitadel/zitadel#5105] [Ideas]
Currently we support only customizing texts of emails/notifications. With Actions we could maybe allow users to send their own emails instead.

Devs could trigger their email-service webhook-style and send out custom mails with the info that we provide from ZITADEL. This would also avoid overly complicated email-templating features in console.

This is an addition to https://github.com/zitadel/zitadel/issues/4143, allowing devs to use an external system completely to design and send mails.

Considerations
- Make a hook available for all notification events
- Make it possible to supress the default notification (eg, password reset email)
- Expose relevant information in the Action (eg, link to password reset) to be used in own code

**Comment by LufyCZ** [comment]
What about being able to define a custom html template? I've just spent 2 hours trying to figure out why the email I sent with a custom `urlTemplate` doesn't look like the html generated (as text) on my backend. Just now I figured out that the `urlTemplate` isn't a template for a URL that generates html for the email, but for that it's a template for the Verify button. Whoops.

**Reply by spietras** [reply]
What about this?

https://github.com/zitadel/zitadel/blob/25d921b20c5cfe8e243d441e86f6a9dd903b6bbe/cmd/defaults.yaml#L993

It is undocumented, but if you are self-hosting it seems to work if you set it before initialising the first instance. But don't know how to change it afterwards, other than messing with the database manually.
